src/org/parosproxy/paros/security/SslCertificateServiceImpl.java

```diff
@@ -142,7 +142,24 @@ public KeyStore createCertForHost(String hostname) throws NoSuchAlgorithmExcepti
 
 		certGen.addExtension(Extension.subjectKeyIdentifier, false, new SubjectKeyIdentifier(pubKey.getEncoded()));
 		certGen.addExtension(Extension.basicConstraints, false, new BasicConstraints(false));
-		certGen.addExtension(Extension.subjectAlternativeName, false, new GeneralNames(new GeneralName(GeneralName.dNSName, hostname)));
+
+		ArrayList<GeneralName> subjectAlternativeNames = new ArrayList<>();
+		boolean subjectAlternativeNamesIsCritical = true;
+		if (hostname != null) {
+			subjectAlternativeNames.add(new GeneralName(GeneralName.dNSName, hostname));
+			subjectAlternativeNamesIsCritical = false;
+		}
+
+		if(!listeningAddress.isAnyLocalAddress()){
+			subjectAlternativeNames.add(new GeneralName(GeneralName.iPAddress, listeningAddress.getHostAddress()));
+		}
+
+		if (hostname == null && listeningAddress.isAnyLocalAddress()) {

```

Comment:
This validation could be done earlier (although see previous comment about wildcard address).